### PR TITLE
Fix n-bodies stack-overflow error with new type Bodies

### PR DESCRIPTION
**Pull request by b-fg**
New `AutoBodies` type is implemented that can solve the n-bodies problem reported in https://github.com/weymouth/WaterLily.jl/issues/88. It holds a vector of type `AutoBody` and implements superposition of `sdf` and `map` functions by iteration instead of recursion. 

Tests show same results as the recursive implementation of the superposition, while n-bodies can be handled now without hitting the stackoverflow error. Also, the measure function is much  faster than before when superposing more than 5 bodies. The cost of the iterative method does not really scale when adding more bodies, while the recursive method seems to scale exponentially.

Here are some test when adding number of bodies `n=6` and `n=11` (max. I can handle in my computer with recursion)
```julia
radius = rand(n)
circles = [(x,t) -> √sum(abs2,x)-r for r ∈ radius]
body = AutoBody(circles[1])
for i ∈ 2:n
    body += AutoBody(circles[i])
end
bodies = AutoBodies([AutoBody(c) for c ∈ circles], +)
xy = rand(2)
```
n=6
```
@btime measure($body, $xy, 1.)
# 5.436 μs (25 allocations: 22.44 KiB)
# 3.670 μs (57 allocations: 2.05 KiB)
```
n=11
```
@btime measure($bodies, $xy, 1.)
# 549.630 μs (25 allocations: 4.96 MiB)
# 3.735 μs (63 allocations: 2.20 KiB)
```

I do not particularly like that there is currently `AutoBody` and `AutoBodies`, but I did not want to get rid/break the previous implementation. Maybe this can be merged in a nicer way though.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -309,7 +309,7 @@ end
         return SA[x+x^3*κ^2/6,y-x^2*κ/2]
     end
     # Test sim_time, and sim_step! stopping time
-    sim = Simulation(radius.*(4,4),(1,0),radius; body=AutoBody(circle), ν, T)
+    sim = Simulation(radius.*(4,4),(1,0),radius; body=AutoBody(circle), ν, T, exitBC=true)
```
Comment (b-fg) [inline test/runtests.jl:312]: I had to update this test again since it was failing locally on 1.10.0.

**Comment by weymouth** [comment]
Did we figure out the test/version/MacOS thing?

Gabriel D Weymouth
_______________________________________________
"Computers are useless. They can only give you answers." Pablo Picasso

File: test/runtests.jl (hunk 1 above)
Comment (weymouth) [reply test/runtests.jl:312]: This is the issue we're seeing in #98 again?

**Comment by b-fg** [comment]
Sadly I cannot test on MacOS, so I am not sure. But the fix on that test is because it fails in Julia 1.10.0 for any OS.

Other than this, the PR is mostly finished. The main new features of this PR is that `Bodies` can account for many `AutoBody` and perform `measure` without maxing out the stack. Also, individual body forces can now be extracted using
```julia
force = WaterLily.∮nds(sim.flow.p,sim.flow.f,sim.body.bodies[i],t)
```

**Comment by b-fg** [comment]
Yes, exactly.

File: test/runtests.jl (hunk 1 above)
Comment (b-fg) [reply test/runtests.jl:312]: Yes, exactly.

**Comment by b-fg** [comment]
Ready for review.

**Comment by b-fg** [comment]
Should we merge this or just close it? @weymouth

**Comment by weymouth** [comment]
We should merge it.

**Pull request merged**